### PR TITLE
[Frame] Reframe polish: sticky manager and offset

### PR DESCRIPTION
**Pull request by sophschneider**
### WHY are these changes introduced?

Part of [#1470](https://github.com/Shopify/polaris-internal/issues/115)

### WHAT is this pull request doing?

1. Accounts for `Frame` offset prop behind `dynamicTopBarAndReframe` project
2. Update `AppProvider` to pass the new reframe scroll container to the `stickyManager`
3. Fixes popover scrolls by wrapping frame content in a `Scrollable`

### How to 🎩
https://admin.web.frame-polish.sophie-schneider.us.spin.dev/store/shop1

1. `Frame` [offset story with feature flag on](https://5d559397bae39100201eedc1-ygmdwznspd.chromatic.com/?path=/story/all-components-frame--with-an-offset&globals=dynamicTopBarAndReframe:!true)
2. Check on s[pin with beta flag on that index tables](https://admin.web.frame-polish.sophie-schneider.us.spin.dev/store/shop1/products/1) are sticking to top of reframe
  - Scroll to bottom of page to variants and the header should stick, adjust window height
3. Check "Skip to content" button with beta flag off and on
4. Check that popovers scroll with content
5. Check no layout shift for when scrollbar is hidden/shows with scrollbar set to "always" in settings

### 🎩 checklist

- [x] Tested a [snapshot](https://github.com/Shopify/polaris/blob/main/documentation/Releasing.md#-snapshot-releases)
- [x] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [x] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)

### CODE REVIEW

File: polaris-react/src/components/Frame/Frame.tsx [hunk 1]
```diff
@@ -73,7 +73,9 @@ interface State {
   showContextualSaveBar: boolean;
 }
 
+const APP_FRAME = 'AppFrame';
 const APP_FRAME_MAIN = 'AppFrameMain';
+const APP_FRAME_CONTENT = 'AppFrameContent';
```
Comment (sophschneider) [inline polaris-react/src/components/Frame/Frame.tsx:78]: Do you think I should export these CONSTs or its ok to copy it into other files?

**Comment by kyledurand** [comment]
Looks like we may need to adjust content spacing as well:


https://github.com/Shopify/polaris/assets/6844391/c33e04ec-5e5e-4eaa-8f50-4cc9636360b8

**Comment by kyledurand** [comment]
Seeing some popover funk. Not sure if spin was updated properly

https://github.com/Shopify/polaris/assets/6844391/bf57ec11-e0d8-4dab-86bd-08e64c33b65c

**Comment by sophschneider** [comment]
@kyledurand thanks! Yeah I saw the popover thing too, I think its because the scroll container used to be on document and now its not falling back properly because the whole app isn't wrapped in a scrollable. I'll try and fix it in this PR! Would probably be worth it for me to do a full `document` audit in polaris to see what else relies on the `document` scroll container

**Comment by sophschneider** [comment]
moving this back to draft as I fix the above things! ill retag for review after

File: polaris-react/src/components/Frame/Frame.module.css [hunk 2]
```diff
@@ -328,14 +313,18 @@
       /* stylelint-disable-next-line polaris/conventions/polaris/custom-property-allowed-list -- private token from component */
       width: calc(
         100vw - var(--pg-navigation-width) -
-          var(--pc-app-provider-scrollbar-width) - var(--p-space-150) -
-          var(--pc-sidebar-width)
+          var(--pc-app-provider-scrollbar-width) - var(--pc-sidebar-width) -
+          var(--p-space-050)
```
Comment (sophschneider) [inline polaris-react/src/components/Frame/Frame.module.css:317]: the `var(--p-space-050)` adds a gap between the scrollbar and the edge of the content so the rounded corner doesn't cut it off.

I may undo this in https://github.com/Shopify/polaris-backlog/issues/1379 when I custom style the scrollbar.

The difference in scrollbar space is also what was causing the layout shifts when turning the beta flag off and on as per [this comment](https://github.com/Shopify/polaris/pull/11945#issuecomment-2077654707).

File: polaris-react/src/components/Frame/Frame.tsx [hunk 3]
```diff
@@ -315,22 +323,25 @@ class FrameInner extends PureComponent<CombinedProps, State> {
...
-                <div
-                  className={classNames(
-                    styles.Content,
-                    hasDynamicTopBar && styles['Content-TopBarAndReframe'],
-                  )}
-                >
-                  {hasDynamicTopBar ? (
+                {hasDynamicTopBar ? (
+                  <Scrollable
+                    scrollbarWidth="thin"
```
Comment (sophschneider) [inline polaris-react/src/components/Frame/Frame.tsx:323]: The scrollbar is currently cut off by the rounded corners of the frame, but I will take a look at that in https://github.com/Shopify/polaris-backlog/issues/1379 when I restyle the scrollbar

File: polaris-react/src/components/Frame/Frame.module.css [hunk 4]
```diff
@@ -435,4 +431,37 @@
...
+    transition: none;
+  }
+
+  @media (--p-breakpoints-md-up) {
+    /* stylelint-disable-next-line polaris/conventions/polaris/custom-property-allowed-list -- private token from component */
+    width: calc(100% - var(--pc-frame-offset, 0px));
+  }
+
+  .hasSidebar & {
+    transition: width var(--p-motion-duration-250) var(--p-motion-ease);
```
Comment (sophschneider) [inline polaris-react/src/components/Frame/Frame.module.css:448]: I have an issue https://github.com/Shopify/polaris-backlog/issues/1606 to update these to transforms for performance

**Comment by sophschneider** [comment]
@kyledurand theres still a bit of a layout shift but its due to the changes in scroll gutter, it should be just a bit!

**Review by kyledurand — APPROVED** [review]
Code and 🎩 look good! I'm slightly worried about changing the shadow bevel wrapper to fixed positioning but I couldn't find any issues 👍

**Review by laurkim — APPROVED** [review]
💯

**Comment by sophschneider** [comment]
@kyledurand yeah I had to do that because scrolling to anchor tags was causing some janky behaviour moving the shadow bevel container. ill let you know if I see any issues with it!

**Pull request merged**